pkg/autodiscovery/listeners/kube_services.go

```diff
@@ -46,6 +47,24 @@ type KubeServiceService struct {
 	creationTime integration.CreationTime
 }
 
+// KubeEndpointListener listens to kubernetes endpoint creation
+type KubeEndpointListener struct {
```

Comment:
Keeping two separate listeners makes each clearer for now, but I'm concerned about how we will implement the following use case:
  - don't send AD services for endpoints that won't match a template (no `/endpoints.` annotations on the associated service)
  - when a annotation is added to the service (by editing it, not removing/recreating it), I want endpoints to be emitted as service, like we do now for service objects.

Could you please have a look at what that would require to add to the KubeEndpointListener ?

Reply:
working on it 👍 :)

Reply:
regarding the first point, in `createEndpoint` we return if an endpoint has not AD annotations
```
if !isEndpointAnnotated(kendpt) {
		// Ignore endpoints with no AD annotation
		return
	}
```
isn't that enough?

Reply:
regarding the second point, since k8s endpoints and services are synced, an annotations edit for a service should trigger the same changes for the related endpoints and which will be handled by our `KubeEndpointListener` `updated` function which creates endpoint service
please let me know if I am misunderstanding the issue :)

Reply:
- The annotation will not be on the endpoint object, but on the associated service object, you'll need to retrieve it

- An edit of the service's annotation will not automatically trigger an update of the endpoints object. Updates can be tracked by looking at an object's `resourceVersion` field, which is the (increasing over time) etcd revision index containing the object version you are looking at.

Here's how to test it: I'm updating the annotation on the service, the service's resVersion is increased, but not the associated endpoints objet's.

```
$ k get svc my-nginx -o yaml |grep resourceVersion | head -n 1
  resourceVersion: "19203976"
$ k get ep my-nginx -o yaml |grep resourceVersion | head -n 1
  resourceVersion: "18995784"
$ k edit svc my-nginx     (modify the `instance` annotation to change a flag, and save)
service/my-nginx edited
$ k get svc my-nginx -o yaml |grep resourceVersion | head -n 1
  resourceVersion: "19204162"
$ k get ep my-nginx -o yaml |grep resourceVersion | head -n 1
  resourceVersion: "18995784"
```

This means the endpoints informer will not trigger an update, and we'll need to trigger a refresh of the endpoints objects based on relevant updates on the service objects. That's the tricky part of the feature :/

Reply:
thanks again for the detailed explanation